cmake/Modules/FindCythonize.cmake

```diff
@@ -0,0 +1,30 @@
+# Find the Cythonize tool.
+#
+# This code sets the following variables:
+#
+#  Cythonize_EXECUTABLE
+#
+# adapted from https://github.com/cmarshall108/cython-cmake-example/blob/master/cmake/FindCython.cmake
+#=============================================================================
+
+if(CMAKE_VERSION VERSION_LESS 3.12)
+    find_package(PythonInterp 3.6 QUIET) # Deprecated since version 3.12
```

Comment:
We are implicitly requiring python-3.6 here, is that wanted?

Reply:
yes

Reply:
Probably we want python >=3.5, although for certain aspects >=3.6 might be preferable. It is definitely not constrained to exactly 3.6. I have mostly been testing on 3.7. Technically cython should work on 3.3+

Reply:
Somewhere in the comments or docs it said Python > 3.5 was required. So that is what I implemented.
For your reference: Our lower boundary for target distributions we want to support is CentOS 7, which comes with Python 3.6. Ubuntu 16.04 has Python 3.5, but we are not testing on it (anymore) and we don't support CMake on it, as the CMake version is too old. So version Python 3.6 is the *de facto* lower boundary for Python 3 (and 2.7 the lower and upper boundary on Python 2)

Reply:
> Probably we want python >=3.5, although for certain aspects >=3.6 might be preferable. It is definitely not constrained to exactly 3.6. I have mostly been testing on 3.7. Technically cython should work on 3.3+

FYI, the version number specified with `find_package()` is the lower boundary. So we can require Python 3.6 when using CMake to configure (which has its lower bound at version 3.10) and not apply any additional restrictions to what we already have.

Reply:
Okay, makes sense -- I tried digging through the cmake docs but couldn't figure out if it was an exact or a lower bound.

Reply:
Actually we only check if python is >=3.6 not cythonize ifself. CMake could still pick up an earlier version as `_python_path` is only used a hints. Is there a required cython version? I only test this PR with cython 0.29.21

Reply:
We could use https://github.com/espressomd/espresso/blob/python/cmake/FindCython.cmake#L25-L30 to check the version.